pandas/tests/indexes/test_base.py

```diff
@@ -240,58 +228,59 @@ def test_constructor_int_dtype_float(self, dtype):
     def test_constructor_int_dtype_nan(self):
         # see gh-15187
         data = [np.nan]
-        msg = "cannot convert"
-
-        with tm.assert_raises_regex(ValueError, msg):
-            Index(data, dtype='int64')
-
-        with tm.assert_raises_regex(ValueError, msg):
-            Index(data, dtype='uint64')
-
-        # This, however, should not break
-        # because NaN is float.
         expected = Float64Index(data)
         result = Index(data, dtype='float')
         tm.assert_index_equal(result, expected)
 
-    def test_index_ctor_infer_nan_nat(self):
+    @pytest.mark.parametrize("dtype", ['int64', 'uint64'])
+    def test_constructor_int_dtype_nan_raises(self, dtype):
+        # see gh-15187
+        data = [np.nan]
+        msg = "cannot convert"
+        with tm.assert_raises_regex(ValueError, msg):
+            Index(data, dtype=dtype)
+
+    @pytest.mark.parametrize("klass,dtype,na_val", [
+        (pd.Float64Index, np.float64, np.nan),
+        (pd.DatetimeIndex, 'datetime64[ns]', pd.NaT)
+    ])
+    def test_index_ctor_infer_nan_nat(self, klass, dtype, na_val):
         # GH 13467
-        exp = pd.Float64Index([np.nan, np.nan])
-        assert exp.dtype == np.float64
-        tm.assert_index_equal(Index([np.nan, np.nan]), exp)
-        tm.assert_index_equal(Index(np.array([np.nan, np.nan])), exp)
-
-        exp = pd.DatetimeIndex([pd.NaT, pd.NaT])
-        assert exp.dtype == 'datetime64[ns]'
-        tm.assert_index_equal(Index([pd.NaT, pd.NaT]), exp)
-        tm.assert_index_equal(Index(np.array([pd.NaT, pd.NaT])), exp)
-
-        exp = pd.DatetimeIndex([pd.NaT, pd.NaT])
-        assert exp.dtype == 'datetime64[ns]'
-
-        for data in [[pd.NaT, np.nan], [np.nan, pd.NaT],
-                     [np.nan, np.datetime64('nat')],
-                     [np.datetime64('nat'), np.nan]]:
-            tm.assert_index_equal(Index(data), exp)
-            tm.assert_index_equal(Index(np.array(data, dtype=object)), exp)
-
-        exp = pd.TimedeltaIndex([pd.NaT, pd.NaT])
-        assert exp.dtype == 'timedelta64[ns]'
-
-        for data in [[np.nan, np.timedelta64('nat')],
-                     [np.timedelta64('nat'), np.nan],
-                     [pd.NaT, np.timedelta64('nat')],
-                     [np.timedelta64('nat'), pd.NaT]]:
-            tm.assert_index_equal(Index(data), exp)
-            tm.assert_index_equal(Index(np.array(data, dtype=object)), exp)
+        na_list = [na_val, na_val]
+        exp = klass(na_list)
+        assert exp.dtype == dtype
+
+        result = Index(na_list)
+        tm.assert_index_equal(result, exp)
+
+        result = Index(np.array(na_list))
+        tm.assert_index_equal(result, exp)
 
+    @pytest.mark.parametrize("pos", [0, 1])
```

Comment:
these tests that are strictly for datetime like things can prob just move to a new file: ``test_datetimelike.py`` (IOW tests that test *all* of DTI,TDI,PI but all together). can be future PR (or here).

IOW test_base should be only non-datetimelike tests.